### DISCUSSION TITLE
More details about usage of data and privacy policy

### DISCUSSION THREAD

**Discussion by tortoiseparrot** [renovatebot/renovate#19553] [Request Help] [answered]
We'd like to use the self-hosted on-premises variant of Mend Renovate.
References:
- https://docs.renovatebot.com/getting-started/running/#mend-renovate-on-premises
- https://github.com/whitesource/renovate-on-prem
- https://www.mend.io/free-developer-tools/renovate/on-premises/

We've sent the following to support@mend.io and were redirected to the discussions here.

Our original questions are as follows:
- Where can we read its EULA, ToS, privacy policy?
- Which PII is being used exactly (e.g. individual user names, emails, IP addresses)?
- Can we influence whether aforementioned PII will be stored, processed, etc. within the European Union exclusively?
- Which external domains and protocols need to be allowed for firewalled internet access?
- Which data is being used in general and how by whom (e.g. PII, internal confidential project names, dependency names and versions – sent to Mend for analytics, to package registries for updated version info)?
- Which parts of the source code do you and third parties require usage rights for (e.g. just the dependency names and versions – no project metadata nor Java code)?
- Which configuration options exist to opt-out of telemetry and other optional data collection and usage?
- How do answers to the above questions differ for the commercial [Mend Remediate](https://docs.renovatebot.com/getting-started/running/#mend-remediate) and does it support self-hosted GitLab instances?

Existing discussions seem to be #8036 which partially answers the questions, but please correct and extend!

- Privacy policy is at https://www.whitesourcesoftware.com/free-developer-tools/terms-of-use#renovate
- It's still unclear which data is being used exactly. Even if this remains on-premises, GDPR and other policies require managing the lifecycle of such data internally
- It seems like data is mostly stored within the on-premises instance only, but see below
- External domains and protocols are still unclear. Certainly at least HTTP APIs of external package registries? Can some of those go through a proxy such as Artifactory for Maven Central and Harbor for Docker/OCI registries (which don't cover all available APIs)?
- What is sent to external package registries in detail is still unclear. Certainly at least package names and versions? What about the names of our own internal dependencies?
- "Usage rights" in particular are tricky since customer contracts might require exclusive usage rights to the code. It seems like Renovate merely requires usage rights for the dependency info extracts of source code?
- It seems like there is no telemetry configuration, but the on-premises instance doesn't phone home anyways, but see above
- It's unclear what the difference between Renovate and Remediate are. The latter is a cloud-hosted version of the former with more integration into other Mend tooling?

**Comment by rarkins** [comment]
Generally speaking we don't dedicate any resources to answering these types of questions for free users because it would take away from our time/budget for open source. I'd encourage you to use the open source version instead so that you can inspect everything for yourself.

**Reply by tortoiseparrot** [reply]
I understand that answering this in detail would take quite some effort if it hasn't been documented before.
Would it be possible to pay you to answer the questions even for the open source version?

Given that there seems to be no telemetry sent to Mend, would you agree that Renovate in the spirit of GDPR only sends the absolutely required data for the task of checking for updates to the third party package repositories?
I understand that this reply would not be legally binding, but your current terms of use are a bit too vague about what (personal) data is used and sent where both for me as a developer and our legal department and thus prevent using it (and a paid instance hosted by you even moreso).

Maybe you could add a sentence like your previous reply in https://github.com/renovatebot/renovate/discussions/8036#discussioncomment-619876 for others to find in your terms of service
> Neither the fully open source CLI or WhiteSource Renovate On-Premises edition send any repository or dependency data home or utilise any hosted service we run.

Basically tl;dr: nothing no tracking nor PII sent to Mend, dependency update info requested from third party package repositories as required for Renovate to do its job

BTW the `#renovate` anchor doesn't seem to exist in the terms of service anymore so it's a bit unclear which parts of it apply to which software.

**Reply by rarkins** [reply]
Your understanding appears to be correct. I will elaborate below.

Let's start with Renovate OSS, which is at the core. It sends no telemetry, sends only the absolutely required data for the task of checking for updates from repositories. It can operate disconnected from the internet e.g. talking only to Artifactory.

Renovate On-Prem is a proprietary but free wrapper of Renovate OSS, so it shares these same characteristics. Although per my memory the terms and conditions say that telemetry may be collected for the purposes of improving the product etc., today it does not - there's no "phone home" for any reason including license validation (which is done locally).

In general I would say that neither product cares about or sends personal data, although of course in some package ecosystems you might see usernames _of packages_ in the names, such as `@rarkins/test` on npm, or `github.com/rarkins/test` on github.

**Comment by tortoiseparrot** [comment, marked as answer]
Quoting https://github.com/renovatebot/renovate/discussions/19553#discussioncomment-4487791 to mark this as answer:
> Your understanding appears to be correct. I will elaborate below.
> 
> Let's start with Renovate OSS, which is at the core. It sends no telemetry, sends only the absolutely required data for the task of checking for updates from repositories. It can operate disconnected from the internet e.g. talking only to Artifactory.
> 
> Renovate On-Prem is a proprietary but free wrapper of Renovate OSS, so it shares these same characteristics. Although per my memory the terms and conditions say that telemetry may be collected for the purposes of improving the product etc., today it does not - there's no "phone home" for any reason including license validation (which is done locally).
> 
> In general I would say that neither product cares about or sends personal data, although of course in some package ecosystems you might see usernames _of packages_ in the names, such as `@rarkins/test` on npm, or `github.com/rarkins/test` on github.

Thanks for taking the time to elaborate @rarkins
